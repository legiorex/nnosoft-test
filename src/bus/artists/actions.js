// Types
import { types } from "./types";

// Instruments
export const artistsActions = {

    // Sync

    fillPosts: (artists) => {
        return {
            type:    types.FILL_POSTS,
            payload: artists,
        };
    },

    // Async

    fetchArtists: () => {
        return {
            type: types.FETCH_ARTISTS_ASYNC,
        };
    },
};
