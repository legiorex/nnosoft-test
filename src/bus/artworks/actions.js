// Types
import { types } from "./types";

export const artworksActions = {

    // Sync

    fillArtworks: (artworks) => {
        return {
            type:    types.FILL_ARTWORKS,
            payload: artworks,
        };
    },

    // Async

    fetchArtworks: () => {
        return {
            type: types.FETCH_ARTWORKS_ASYNC,
        };
    },
};
