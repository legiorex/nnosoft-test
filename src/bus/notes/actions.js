// Types
import { types } from "./types";

export const notesActions = {

    // Sync

    addNote: (note) => {
        return {
            type:    types.ADD_NOTE,
            payload: note,
        };
    },

    writeNotesLocal: () => {
        return {
            type: types.WRITE_NOTES_LOCAL,
        };
    },
    sendNotes: (notes) => {
        return {
            type:    types.SEND_NOTES,
            payload: notes,
        };
    },

};
