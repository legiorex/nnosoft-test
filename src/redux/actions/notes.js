export const ADD_NOTE = "ADD_NOTE";

export function addNote (text, afterIndex) {
    return {
        type: ADD_NOTE,
        text,
        afterIndex,
    };
}
