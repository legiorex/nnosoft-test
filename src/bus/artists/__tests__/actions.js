// Actions
import { artistsActions } from '../actions';

describe(' artists actions', () => {
    test('fillArtists', () => {
        expect(artistsActions.fillArtists(__.artists)).toMatchSnapshot();
    });

    test('fetchArtists', () => {
        expect(artistsActions.fetchArtists()).toMatchSnapshot();
    });
});
