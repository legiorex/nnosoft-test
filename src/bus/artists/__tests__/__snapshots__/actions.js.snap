// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` artists actions fetchArtists 1`] = `
Object {
  "type": "FETCH_ARTISTS_ASYNC",
}
`;

exports[` artists actions fillArtists 1`] = `
Object {
  "payload": Array [
    Object {
      "_id": "1",
      "firstName": "Annie",
      "lastName": "Walker",
      "photo": "https://www.url.com",
    },
  ],
  "type": "FILL_ARTISTS",
}
`;
